src/worker/helpers.go

```diff
@@ -166,46 +166,192 @@ func initOLDir(olPath string, dockerBaseImage string, newBase bool) (err error)
 	return nil
 }
 
-// stop OL if running (or just return if not).
-// main error scenarios:
-// 1. PID exists, but process cannot be killed (worker probably died unexpectedly)
-// 2. The cleanup is taking too long (maybe the timeout is insufficient, or there is a deadlock)
-func stopOL(_ string) error {
-	// locate worker.pid, use it to get worker's PID
+// Define a custom type for the state
+type OlState int
+
+// Define constants for the different states
+const (
+	Stopped OlState = iota
+	Running
+	DirtyShutdown
```

Comment:
Perhaps the relationship between Stopped and DirtryShutdown could be clearer.  Maybe these names?
StoppedClean
StoppedDirty